cob_people_detection/ros/launch/people_detection.launch

```diff
@@ -18,36 +22,69 @@ for Care-O-bot default is cam3d_nodelet_manager -->
+       <arg name="colorimage_in_topic" value="$(arg colorimage_in_topic)"/>
+       <arg name="pointcloud_rgb_in_topic" value="$(arg pointcloud_rgb_in_topic)"/>
+     </include>
+   </group>
+   <group if="$(arg using_nodelets)">
+     <!-- special launch files for usage with nodelets -->
+     <node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>
+     <include file="$(find cob_people_detection)/ros/launch/sensor_message_gateway_nodelet.launch">
+       <arg name="nodelet_namespace" value="$(arg camera_namespace)"/>
+       <arg name="nodelet_manager" value="$(arg nodelet_manager)"/>
+       <arg name="colorimage_in_topic" value="$(arg colorimage_in_topic)"/>
+       <arg name="pointcloud_rgb_in_topic" value="$(arg pointcloud_rgb_in_topic)"/>
+     </include>
+   </group>
+
+  <!-- remaining launch files -->
+  <param name="data_storage_directory" type="string" value="$(env HOME)/.ros/cob_people_detection/files/"/>
+  <param name="face_detector_method" type="int" value="$(arg face_detector_method)"/> <!-- 0 -> hog: Histogram of Oriented Gradients,
+                                                              1 -> vj: viola-jones -->
+  <!--Head detector-->
+  <include if="$(arg launch_head_detector)" file="$(find cob_people_detection)/ros/launch/head_detector.launch"/>
+
+  <!--Face detector-->
+  <group if="$(arg launch_face_detector)">
+    <include if="$(arg face_detector_method)" file="$(find cob_people_detection)/ros/launch/face_detector_hog.launch"/>
+    <include unless="$(arg face_detector_method)" file="$(find cob_people_detection)/ros/launch/face_detector.launch"/>
   </group>
-  <group if="$(arg using_nodelets)">
-    <!-- special launch files for usage with nodelets -->
-    <node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>
-    <include file="$(find cob_people_detection)/ros/launch/sensor_message_gateway_nodelet.launch">
-      <arg name="nodelet_namespace" value="$(arg camera_namespace)"/>
-      <arg name="nodelet_manager" value="$(arg nodelet_manager)"/>
-      <arg name="colorimage_in_topic" value="$(arg colorimage_in_topic)"/>
-      <arg name="pointcloud_rgb_in_topic" value="$(arg pointcloud_rgb_in_topic)"/>
+
+  <!--Face recognition-->
+  <group if="$(arg launch_face_recognizer)">
+    <include file="$(find cob_people_detection)/ros/launch/online_face_recognizer.launch">
+      <arg name="recognition_method" value="$(arg recognition_method)"/>
     </include>
+
+    <node pkg="cob_people_detection" type="toggle_online_learning_service_node.py" name="toggle_online_learning_service" output="screen" ns="online_face_recognizer"/>
   </group>
 
-  <!-- remaining launch files -->
-  <param name="data_storage_directory" type="string" value="$(env HOME)/.ros/cob_people_detection/files/"/>
+  <group>
```

Comment:
This should be part of above's group `<group if="$(arg launch_face_recognizer)">`